packages/query/src/block-processor.ts

```diff
@@ -391,17 +423,46 @@ export class BlockProcessor implements BlockProcessorInterface {
     // unnecessary calls to the RPC node for validator infos. Instead, we'll
     // only get updated validator infos once we're within the latest known
     // epoch.
-    if (nextEpochIsLatestKnownEpoch) void this.updateValidatorInfos();
+    if (nextEpochIsLatestKnownEpoch) void this.updateValidatorInfos(nextEpochStartHeight);
   }
 
-  private async updateValidatorInfos(): Promise<void> {
+  private async updateValidatorInfos(nextEpochStartHeight: bigint): Promise<void> {
     for await (const validatorInfoResponse of this.querier.staking.allValidatorInfos()) {
       if (!validatorInfoResponse.validatorInfo) continue;
 
       // Await the upsert. This makes it possible for users of this method to
       // await the entire method, if they want to block all other code until all
       // validator infos have been upserted.
       await this.indexedDb.upsertValidatorInfo(validatorInfoResponse.validatorInfo);
+
+      // Don't await this, though -- price equivalents for delegation tokens are
+      // non-critical, and shouldn't block the rest of the block processor.
+      void this.updatePriceForValidatorDelegationToken(validatorInfoResponse, nextEpochStartHeight);
```

Comment:
In reality though, this will simply run when the call stack is empty right? In between loops of the `compactBlockRange()` iterator? If so, it probably doesn't make a difference?

This is one of those moments where I wish it were easier to just throw this in another thread and forget it.

Reply:
Sort of... `.updatePriceForValidatorDelegationToken()`has a few `await`s of its own, so I guess those would get threaded between following frames of JS work.